cli/clistat/stat.go

```diff
@@ -34,38 +82,29 @@ func (r *Result) String() string {
 		return "-"
 	}
 
-	var usedDisplay, totalDisplay string
-	var usedScaled, totalScaled float64
-	var usedPrefix, totalPrefix string
-	usedScaled, usedPrefix = humanize.ComputeSI(r.Used)
-	usedDisplay = humanizeFloat(usedScaled)
-	if r.Total != (*float64)(nil) {
-		totalScaled, totalPrefix = humanize.ComputeSI(*r.Total)
-		totalDisplay = humanizeFloat(totalScaled)
+	scale := 1.0
+	if r.Prefix != 0.0 {
+		scale = float64(r.Prefix)
 	}
 
 	var sb strings.Builder
-	_, _ = sb.WriteString(usedDisplay)
-
-	// If the unit prefixes of the used and total values are different,
-	// display the used value's prefix to avoid confusion.
-	if usedPrefix != totalPrefix || totalDisplay == "" {
-		_, _ = sb.WriteString(" ")
-		_, _ = sb.WriteString(usedPrefix)
-		_, _ = sb.WriteString(r.Unit)
-	}
-
-	if totalDisplay != "" {
+	var usedScaled, totalScaled float64
+	usedScaled = r.Used / scale
+	_, _ = sb.WriteString(humanizeFloat(usedScaled))
+	if r.Total != (*float64)(nil) {
 		_, _ = sb.WriteString("/")
-		_, _ = sb.WriteString(totalDisplay)
-		_, _ = sb.WriteString(" ")
-		_, _ = sb.WriteString(totalPrefix)
-		_, _ = sb.WriteString(r.Unit)
+		totalScaled = *r.Total / scale
+		_, _ = sb.WriteString(humanizeFloat(totalScaled))
 	}
 
-	if r.Total != nil && *r.Total != 0.0 {
+	_, _ = sb.WriteString(" ")
+	_, _ = sb.WriteString(r.Prefix.String())
+	_, _ = sb.WriteString(r.Unit)
+
+	if r.Total != (*float64)(nil) && *r.Total > 0 {
```

Comment:
Is the typed nil check really necessary here? I would expect this to be writable as `r.Total != nil`.

Reply:
I found it to be necessary, otherwise the unit tests fail.